src/protocols/secure_channel/RendezvousParameters.h

```diff
@@ -128,9 +173,10 @@ class RendezvousParameters
     }
 
 private:
-    Transport::PeerAddress mPeerAddress;  ///< the peer node address
-    uint32_t mSetupPINCode  = 0;          ///< the target peripheral setup PIN Code
-    uint16_t mDiscriminator = UINT16_MAX; ///< the target peripheral discriminator
+    Transport::PeerAddress mPeerAddress; ///< the peer node address
+    uint32_t mSetupPINCode = 0;          ///< the target peripheral setup PIN Code
+    bool mHasDiscriminator = false;
+    SetupDiscriminator mSetupDiscriminator;
```

Comment:
Any value in using `std::optional<SetupDiscriminator>' here?

Reply:
Done